### PR TITLE
refactor: centralize version and version-range

### PR DESCRIPTION
**Pull request by jkowalleck**
goal: improve documentation by consolidating elements regarding "version".
pure refactoring, no new functionality was added, nor removed, nor changed.

- [x] consolidate `version`
- [x] consolidate `range`
- [x] rename `range` definition to `version-range`
- [x] add more examples
- [x] review rendered documentation

### CODE REVIEW

File: schema/bom-1.6.schema.json [hunk 1]
```diff
@@ -2320,16 +2318,36 @@
...
+      "maxLength": 1024,
+      "examples": [
+        "9.0.14",
+        "1.2.3|>=2.0.0|<5.0.0",
+        "0.0.0|0.0.1|0.0.2|0.0.3|1.0|2.0pre1",
+        ">=1.0.0-beta1|<=1.7.5|>=7.0.0-M1|<=7.0.7|>=7.1.0|<=7.1.2|>=8.0.0-M1|<=8.0.1",
+        ">=2.2.0|!= 2.2.1|<2.3.0"
+      ]
+    },
+    "range": {
```
Comment (jkowalleck) [inline schema/bom-1.6.schema.json:2347]: the name was too broad. therefore, it needed to be renamed to `version-range`

BUT `range` was en exisitn definition, which may have been used/referenced in JSON schemas downstream, by organizations that build supersets on out schema. 
Therefore, we cannot simply remove/rename it. 

SO it stayed as is, but is marked as deprecated and "inherits" from the new definition.

File: schema/bom-1.6.schema.json [hunk 2]
```diff
@@ -2320,16 +2318,36 @@
...
+        "9.0.14",
+        "v1.33.7",
+        "7.0.0-M1",
+        "2.0pre1",
+        "1.0.0-beta1",
+        "0.8.15"
+      ]
     },
-    "range": {
+    "version-range": {
```
Comment (stevespringett) [inline schema/bom-1.6.schema.json:2334]: If I'm not mistaken, we use lower camel case for multi-word definitions. I'd like to be consistent. Therefore I think it should be `versionRange`.

File: schema/bom-1.6.xsd [hunk 3]
```diff
@@ -54,6 +54,43 @@ limitations under the License.
...
+                - "7.0.0-M1"
+                - "2.0pre1"
+                - "1.0.0-beta1"
+                - "0.8.15"
+            ]]>
+            </xs:documentation>
+        </xs:annotation>
+        <xs:restriction base="xs:normalizedString"/>
+    </xs:simpleType>
+    <xs:simpleType name="version-range">
```
Comment (stevespringett) [inline schema/bom-1.6.xsd:74]: Same comment as above regarding lower camel case.

Comment (jkowalleck) [reply schema/bom-1.6.xsd]: thanks. fixed that.

File: schema/bom-1.6.schema.json (hunk 2 above)
Comment (jkowalleck) [reply schema/bom-1.6.schema.json]: you are right. fixed that.

**Comment by jkowalleck** [comment]
@stevespringett thank you for the review. you are right, and I adjusted to the naming convention.

**Pull request merged**